### PR TITLE
feat(ux): expose `rows` and `range` keyword arguments in `value.over()`

### PR DESCRIPTION
**Pull request by kszucs**
depends on #5014

### CODE REVIEW

**Comment by cpcloud** [comment]
@kszucs Can you rebase?

File: ibis/tests/expr/test_window_functions.py [hunk 1]
```diff
@@ -38,3 +38,27 @@ def test_mutate_with_analytic_functions(alltypes):
...
+    expr1 = t.f.cumsum().over(rows=(0, 1), group_by=t.g, order_by=[t.f, t.h])
+    expected1 = t.f.cumsum().over(w1)
+    assert expr1.equals(expected1)
+
+    expr2 = t.f.cumsum().over(range=(-1, 1), group_by=[t.g, t.a], order_by=[t.f])
+    expected2 = t.f.cumsum().over(w2)
+    assert expr2.equals(expected2)
+
+    expr3 = t.f.cumsum().over(
+        between=(ibis.interval(days=1), ibis.interval(days=5)), order_by=[t.f]
```
Comment (kszucs) [inline ibis/tests/expr/test_window_functions.py:61]: I'm not sure that we should expose the between keyword argument, it enables inferring the window frame kind if it isn't passed explicitly.

I tend to think that we should enforce either `over(rows=` or `over(range=)` but not `over(between=)`.

File: ibis/expr/types/groupby.py [hunk 2]
```diff
@@ -222,19 +223,43 @@ def _get_window(self):
                 order_by=self._window.order_by + self._order_by,
             )
 
-    def over(self, window) -> GroupedTable:
+    def over(
```
Comment (kszucs) [inline ibis/expr/types/groupby.py:226]: Actually this method is not used anywhere, at least removing it doesn't break a single test.

Comment (cpcloud) [reply ibis/expr/types/groupby.py:226]: Yeah. I didn't realize we had support for applying a window over a group by. This makes sense to me though. It avoids having to duplicate a window if the rows/range specification are more complex than can be done with `.group_by(...).mutate(...)`

**Review by cpcloud — APPROVED** [review]

**Pull request merged**